diff --git a/buildSrc/private/src/main/kotlin/androidx/build/binarycompatibilityvalidator/BinaryCompatibilityValidation.kt b/buildSrc/private/src/main/kotlin/androidx/build/binarycompatibilityvalidator/BinaryCompatibilityValidation.kt
index 12e2dc75baa..2370a94909b 100644
--- a/buildSrc/private/src/main/kotlin/androidx/build/binarycompatibilityvalidator/BinaryCompatibilityValidation.kt
+++ b/buildSrc/private/src/main/kotlin/androidx/build/binarycompatibilityvalidator/BinaryCompatibilityValidation.kt
@@ -42,7 +42,7 @@ import org.gradle.api.file.RegularFile
 import org.gradle.api.file.RegularFileProperty
 import org.gradle.api.provider.Provider
 import org.gradle.api.tasks.TaskProvider
-import org.jetbrains.kotlin.abi.tools.api.v2.KlibTarget
+import org.jetbrains.kotlin.abi.tools.KlibTarget
 import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
 import org.jetbrains.kotlin.gradle.plugin.KotlinCompilation.Companion.MAIN_COMPILATION_NAME
 import org.jetbrains.kotlin.gradle.plugin.KotlinPlatformType
diff --git a/buildSrc/private/src/main/kotlin/androidx/build/binarycompatibilityvalidator/GenerateAbiTask.kt b/buildSrc/private/src/main/kotlin/androidx/build/binarycompatibilityvalidator/GenerateAbiTask.kt
index c9da66caf83..477cd2f6630 100644
--- a/buildSrc/private/src/main/kotlin/androidx/build/binarycompatibilityvalidator/GenerateAbiTask.kt
+++ b/buildSrc/private/src/main/kotlin/androidx/build/binarycompatibilityvalidator/GenerateAbiTask.kt
@@ -37,9 +37,9 @@ import org.gradle.api.tasks.TaskAction
 import org.gradle.workers.WorkAction
 import org.gradle.workers.WorkParameters
 import org.gradle.workers.WorkerExecutor
-import org.jetbrains.kotlin.abi.tools.AbiToolsFactory
-import org.jetbrains.kotlin.abi.tools.api.AbiFilters
-import org.jetbrains.kotlin.abi.tools.api.v2.KlibTarget
+import org.jetbrains.kotlin.abi.tools.AbiFilters
+import org.jetbrains.kotlin.abi.tools.AbiTools
+import org.jetbrains.kotlin.abi.tools.KlibTarget
 
 @CacheableTask
 abstract class GenerateAbiTask
@@ -78,7 +78,7 @@ abstract class KlibDumpWorker : WorkAction<KlibDumpWorker.Parameters> {
         abstract val excludedAnnotatedWith: SetProperty<String>
     }
 
-    private val abiTools = AbiToolsFactory().get().v2
+    private val abiTools = AbiTools.getInstance()
 
     override fun execute() {
         val klibTargets = parameters.klibs.get()
diff --git a/buildSrc/private/src/main/kotlin/androidx/build/clang/KonanBuildService.kt b/buildSrc/private/src/main/kotlin/androidx/build/clang/KonanBuildService.kt
index f7125b647a3..8d021562aef 100644
--- a/buildSrc/private/src/main/kotlin/androidx/build/clang/KonanBuildService.kt
+++ b/buildSrc/private/src/main/kotlin/androidx/build/clang/KonanBuildService.kt
@@ -146,7 +146,8 @@ abstract class KonanBuildService @Inject constructor(private val execOperations:
                         TempFiles(),
                         objectFiles = objectFiles,
                         executable = outputFile.canonicalPath,
-                        libraries = linkedObjectFiles,
+                        dynamicLibraries = linkedObjectFiles,
+                        staticLibraries = emptyList(),
                         linkerArgs = linkerFlags,
                         optimize = true,
                         debug = false,
